src/main/java/de/tum/in/www1/artemis/service/exam/StudentExamService.java

```diff
@@ -181,8 +181,7 @@ private void saveSubmission(User currentUser, List<StudentParticipation> existin
         // if exercise is either QuizExercise, TextExercise or ModelingExercise and exactly one participation exists
         if (exercise.getStudentParticipations() != null && exercise.getStudentParticipations().size() == 1) {
             for (StudentParticipation studentParticipation : exercise.getStudentParticipations()) {
-                StudentParticipation existingParticipation = existingParticipations.stream().filter(p -> p.getId().equals(studentParticipation.getId()))
-                        .collect(Collectors.toList()).get(0);
+                StudentParticipation existingParticipation = existingParticipations.stream().filter(p -> p.getId().equals(studentParticipation.getId())).toList().get(0);
```

Comment:
```suggestion
                StudentParticipation existingParticipation = existingParticipations.stream().filter(p -> p.getId().equals(studentParticipation.getId())).findFirst().orElseThrow();
```
I think using findFirst here shows more clearly that we only care about a single element (and avoids building a list first).
The `orElseThrow` should be okay here as the `get(0)` would have thrown an exception before if the list was empty as well.

Possibly out-of-scope for this PR: Maybe this pattern is repeated multiple times and could be replaced everywhere?